chain/client/tests/catching_up.rs

```diff
@@ -641,4 +641,153 @@ mod tests {
+            return;
+        }
+        assert!(height_hold > 1);
+        let validator_groups = 1;
+        init_integration_logger();
+        System::run(move || {
+            let connectors: Arc<RwLock<Vec<(Addr<ClientActor>, Addr<ViewClientActor>)>>> =
+                Arc::new(RwLock::new(vec![]));
+
+            let (validators, key_pairs) = get_validators_and_key_pairs();
+
+            let phase = Arc::new(RwLock::new(OrphansPhases::WaitingForSendingTx));
+
+            let connectors1 = connectors.clone();
+            let block_prod_time: u64 = 1200;
+            let (_, conn) = setup_mock_all_validators(
+                validators.clone(),
+                key_pairs.clone(),
+                validator_groups,
+                true,
+                block_prod_time,
+                false,
+                false,
+                5,
+                Arc::new(RwLock::new(move |from_whom: String, msg: &NetworkRequests| {
+                    let account_from = "test3.3".to_string();
+                    let account_to = "test1.1".to_string();
+                    let source_shard_id = account_id_to_shard_id(&account_from, 4);
+                    let destination_shard_id = account_id_to_shard_id(&account_to, 4);
+
+                    let mut phase = phase.write().unwrap();
+                    match *phase {
+                        OrphansPhases::WaitingForSendingTx => {
+                            if let NetworkRequests::Block { block } = msg {
+                                assert!(block.header.inner_lite.height <= height_hold - 1);
+                                if block.header.inner_lite.height == height_hold - 1 {
+                                    println!(
+                                        "From shard: {}, to shard: {}",
+                                        source_shard_id, destination_shard_id,
+                                    );
+                                    for i in 0..16 {
+                                        send_tx(
+                                            &connectors1.write().unwrap()[i].0,
+                                            account_from.clone(),
+                                            account_to.clone(),
+                                            111,
+                                            1,
+                                            block.header.prev_hash,
+                                        );
+                                    }
+                                    *phase = OrphansPhases::HoldingBlock;
+                                }
+                            }
+                        }
+                        OrphansPhases::HoldingBlock => {
+                            // We don't allow to get any chunks from height `height_hold` - 1.
+                            // It means all blocks with higher height will be put into orphans.
+                            if let NetworkRequests::PartialEncodedChunkRequest { .. } = msg {
```

Comment:
This doesn't really test that "test1.2" uses early chunk retrieval, right?
Even without your change, once the other validators collectively get to height `height_release`, "test1.2" will start unorphaning its blocks, and requesting chunks for them serially, eventually syncing with everyone.
What we really want to do here is:

1. Blocks `PartialEncodedChunkRequest` for the height `height_hold - 1`.
2. Check that for each height between `height_hold` and `height_release` (left included right excluded) there was a `PartialEncodedchunkRequest` sent by "test1.2" *before* we switched to validting orphans.